### DISCUSSION TITLE
unique disappear after running introspect (MySQL)

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#7820] [Q&A] [answered]
My schema is 
```
model User {
  id                                       String                       @id @default(cuid())
  email                                    String                       @unique
}
```
My database (MySQL) is 

```
CREATE TABLE `User` (
  `id` char(30) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  `email` mediumtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,
 
  PRIMARY KEY (`id`),
  UNIQUE KEY `id_UNIQUE` (`id`),
  UNIQUE KEY `email_UNIQUE` (`email`(191)),

) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci
```

After running `npx prisma introspect`
My schema became: 
```
model User {
  id                                       String                       @id @default(cuid()) @db.Char(30)
  email                                    String                       @db.MediumText
}
```

Why `unique` disappears? What should I add in the database to keep this constraint in the schema?

**Comment by janpio** [comment, marked as answer]
The problem here is probably the `191` which Prisma does not support yet, it only supports full indexes.

**Reply by alan345** [reply]
Thanks!

 If I do:
```
ALTER TABLE User DROP INDEX `email_UNIQUE`; // to remove the unique index
CREATE UNIQUE INDEX email ON User(email); // to recreate the index without 191
```

I got this error:
```
[Error code:1170 SQL state:42000] BLOB/TEXT column 'email' used in key specification without a key length
```
  Any idea How can I create the full index?

**Reply by janpio** [reply]
You probably can't, that is the problem by this functionality missing.

**Reply by alan345** [reply]
so.. What I am supposed to do to use Prisma? (and to be recognized as a unique field by prisma)?

**Reply by janpio** [reply]
When you run `introspect`, you have to undo that change to the schema again. Everything else should work fine.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
